apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/TableSchemaExtensions.cs

```diff
@@ -40,5 +40,38 @@ internal static class TableSchemaExtensions
             }
             return ret;
         }
+
+        internal static string BuildSelectedFields(this TableSchema schema)
+        {
+            if (!(schema?.Fields?.Count > 0))
```

Comment:
This is where the null handling matters. I would suggest that if `schema` is null, we should return a null string to indicate "all fields". It's only if we've got a non-null schema with null/empty fields that we want to say "don't return any fields" I think - but we should add a clear comment about that, too :)